blockchain/blockchain.go

```diff
@@ -105,20 +119,42 @@ func (b *Blockchain) head(txn db.Transaction) (*core.Block, error) {
 	}
 }
 
+func (b *Blockchain) headsHeader(txn db.Transaction) (*core.Header, error) {
```

Comment:
We dont really need this helper. This can all be done in `HeadsHeader`. Personally I wouldnt expect `headsHeader` to be reused

Reply:
removed